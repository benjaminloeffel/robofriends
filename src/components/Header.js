import { Styled } from "theme-ui";
import PropTypes from "prop-types";
/** @jsx jsx */
import { jsx } from "theme-ui";
import SearchInput from "./SearchInput";

const Header = ({ searchTerms, searchChange, resultLength }) => {
  return (
    <header
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center"
      }}
    >
      <Styled.h1>RoboFriends</Styled.h1>
      <Styled.p>Search and instantly find your Robot Friends!</Styled.p>
      <SearchInput searchTerm={searchTerms} searchChange={searchChange} />
      <Styled.p>{resultLength} Robots found.</Styled.p>
    </header>
  );
};

Header.propTypes = {
  searchTerms: PropTypes.string.isRequired,
  searchChange: PropTypes.func.isRequired,
  resultLength: PropTypes.number.isRequired
};

export default Header;
