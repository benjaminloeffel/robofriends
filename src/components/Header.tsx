/** @jsx jsx */
import React, { memo } from "react";
import { jsx, Styled } from "theme-ui";
import PropTypes from "prop-types";
import SearchInput from "./SearchInput";
import ButtonDarkMode from "./ButtonDarkMode";

type Props = {
  searchTerms: string;
  searchChange(event: React.SyntheticEvent<HTMLInputElement>): void;
};

const Header: React.FC<Props> = ({ searchTerms, searchChange }) => {
  const Heading = memo(() => (
    <React.Fragment>
      <Styled.h1>RoboFriends</Styled.h1>
      <Styled.p>Search and instantly find your Robot Friends!</Styled.p>
    </React.Fragment>
  ));

  return (
    <header
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        bg: "background",
        padding: 16
      }}
    >
      <Heading />
      <SearchInput searchTerm={searchTerms} searchChange={searchChange} />
      <ButtonDarkMode />
    </header>
  );
};

Header.propTypes = {
  searchTerms: PropTypes.string.isRequired,
  searchChange: PropTypes.func.isRequired
};

export default Header;
