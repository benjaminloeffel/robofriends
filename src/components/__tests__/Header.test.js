import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import Header from "../Header";

describe("Header", () => {
  const mockProps = {
    searchTerm: "test",
    searchChange: jest.fn()
  };

  const wrapper = shallow(
    <Header
      searchTerms={mockProps.searchTerm}
      searchChange={mockProps.searchChange}
    />
  );

  it("render the Header component with props", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
