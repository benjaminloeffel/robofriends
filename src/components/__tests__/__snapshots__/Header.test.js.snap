// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render the Header component with props 1`] = `ShallowWrapper {}`;
